include/server.h

```diff
@@ -5,11 +5,19 @@
 
 #include "connection.h"
 
+struct ServerInfo;
+
 class Server {
  public:
-  Server(boost::asio::io_service& io_service, const char* filename);
+  // Factory method
+  static Server *makeServer(boost::asio::io_service& io_service, const char* config_file);
```

Comment:
edit spacing so it's easier to identify "const char* config_file);" should be on the next line